Update readme and add example (#234)
arduino/arduino-iot-js: README.md

please check the official [Javascript Rest API client](https://www.npmjs.com/package/@arduino/arduino-iot-client).

If you want to learn more about Arduino IoT Cloud architecture, check the official [getting started documentation](https://www.arduino.cc/en/IoT/HomePage). 

## Installation

```bash
$ npm install arduino-iot-js
```

## How to use
The MQTT connection over Websocket relies on Username / Password authentication. Under the hood, this module uses your user ID (plus a timestamp) as *Username* and needs a valid JWT Token as *Password*. You can use either a valid JWT token or just your API Credentials (*clientId* and *clientSecret*).

### How to import arduino-iot-js in your project
Using a web application in the browser
```javascript
import ArduinoIoTCloud from 'arduino-iot-js'
```
Using nodejs
```javascript
const { ArduinoIoTCloud } = require('arduino-iot-js');
```

### How to connect to Arduino IoT Cloud broker using API Credentials
```javascript
const { ArduinoIoTCloud } = require('arduino-iot-js');

const options = {
    clientId: "YOUR_CLIENT_ID",
    clientSecret: "YOUR_CLIENT_SECRET",
    onDisconnect: message => {
        console.error(message);
    }
}

> ArduinoIoTCloud.connect(options).then(() => {
>     console.log("Connected to Arduino IoT Cloud broker");
> }, error => {
>     console.error(error);
> });
> ```

Reviewer: ```suggestion
ArduinoIoTCloud.connect(options)
  .then(() => console.log("Connected to Arduino IoT Cloud broker"))
  .catch(error => console.error(error));
```

### How to listen for property value updates
After a successful connection, you can listen for property updates.
To do this you need:
- The ID of the *Thing* the *property* belongs to. You can list all your things and properties using the [Javascript Rest API client](https://www.npmjs.com/package/@arduino/arduino-iot-client), calling the [GET Things endpoint](https://www.arduino.cc/reference/en/iot/api/index.html#api-ThingsV2-thingsV2List)
- The *variable name* of the property you want to listen

```javascript
const { ArduinoIoTCloud } = require('arduino-iot-js');

> const thingID = "THING_ID"

Reviewer: typo
```suggestion
const thingId = "THING_ID"
```

const variableName = "PROPERTY_NAME"

const options = {
    clientId: "YOUR_CLIENT_ID",
    clientSecret: "YOUR_CLIENT_SECRET",
    onDisconnect: message => {
        console.error(message);
    }
}

> ArduinoIoTCloud.connect(options).then(() => {
>     console.log("Connected to Arduino IoT Cloud broker");
>     ArduinoIoTCloud.onPropertyValue(thingId, variableName, showUpdates = value => {
>         console.log(value);
>     }).then(() => {
>         console.log("Callback registered");
>     }, error => {
>         console.log(error);
>     });
> });
> ```

Reviewer: ```suggestion
ArduinoIoTCloud.connect(options)
  .then(() => {
    console.log("Connected to Arduino IoT Cloud broker");
    return ArduinoIoTCloud.onPropertyValue(thingId, variableName,
      value => console.log(value));
  })
  .then(() => console.log("Callback registered"))
  .catch(error => console.log(error));
```

Author: I think that using an anonymous function as property update callback doesn't work

Each time a new value is sent from the Device, the `counterUpdates` callback will be called.

### How to disconnect from Arduino IoT Cloud Broker
```javascript

> ArduinoCloud.disconnect().then(() => {
>   console.log("Successfully disconnected");
> });

Reviewer: ```suggestion
ArduinoCloud.disconnect()
  .then(() => console.log("Successfully disconnected"));
```

```
### How to send property values to the device
To do this you need:
- The ID of the *Thing* the *property* belongs to. You can list all your things and properties using the [Javascript Rest API client](https://www.npmjs.com/package/@arduino/arduino-iot-client),  calling the [GET Things endpoint](https://www.arduino.cc/reference/en/iot/api/index.html#api-ThingsV2-thingsV2List)
- The *variable name* of the property you want to set
- Value can be either a string, a boolean or a number
```javascript
const { ArduinoIoTCloud } = require('arduino-iot-js');

> const thingID = "THING_ID"

Reviewer: typo
```suggestion
const thingId = "THING_ID"
```

const variableName = "PROPERTY_NAME"

const options = {
    clientId: "YOUR_CLIENT_ID",
    clientSecret: "YOUR_CLIENT_SECRET",
    onDisconnect: message => {
        console.error(message);
    }
}

> ArduinoIoTCloud.connect(options).then(() => {
>     console.log("Connected to Arduino IoT Cloud broker");
>     ArduinoCloud.sendProperty(thingId, variableName, value).then(() => {
>         console.log("Property value correctly sent");
>     });    
> });

Reviewer: ```suggestion
ArduinoIoTCloud.connect(options)
  .then(() => {
    console.log("Connected to Arduino IoT Cloud broker");
    return ArduinoCloud.sendProperty(thingId, variableName, value).
      then(() => console.log("Property value correctly sent"));
  })
  .catch(error => console.log(error));
```

```
### How to listen to every user properties updates
```javascript
const { ArduinoIoTCloud } = require('arduino-iot-js');
const ArduinoIoTApi = require('@arduino/arduino-iot-client');

const options = {
    clientId: "YOUR_CLIENT_ID",
    clientSecret: "YOUR_CLIENT_SECRET",
    onDisconnect:  message  => {
        console.error(message);
    }
}

// Connect to Arduino IoT Cloud MQTT Broker

> ArduinoIoTCloud.connect(options).then(() => {
>     console.log("Connected to Arduino IoT Cloud MQTT broker");
> 	
> 	// Init Arduino API Client
>     const  ArduinoIoTClient = ArduinoIoTApi.ApiClient.instance;
>     ArduinoIoTClient.authentications['oauth2'].accessToken = ArduinoIoTCloud.getToken();
> 
>     const thingsApi = new ArduinoIoTAPI.ThingsV2Api(ArduinoIoTClient);
>     const propertiesApi = new  ArduinoIoTApi.PropertiesV2Api(ArduinoIoTClient);
> 
>     thingsApi.thingsV2List().then(things  => {
>         things.forEach(thing  => {
>             propertiesAPI.propertiesV2List(thing.id).then(properties  => {
>             properties.forEach(property  => {
>                 ArduinoIoTCloud.onPropertyValue(thing.id, property.variable_name, update = value  => {
>                     console.log(property.variable_name+": "+value)
>                 }).then(() => {
>                     console.log("Callback registered for "+property.variable_name);
>                 });
>             });
>         });
>     });
> }, error  => {
>     console.log(error)
> });

Reviewer: ```suggestion
ArduinoIoTCloud.connect(options)
  .then(() => {
    console.log("Connected to Arduino IoT Cloud MQTT broker");

    // Init Arduino API Client
    const ArduinoIoTClient = ArduinoIoTApi.ApiClient.instance;
    ArduinoIoTClient.authentications['oauth2'].accessToken = ArduinoIoTCloud.getToken();

    const thingsApi = new ArduinoIoTAPI.ThingsV2Api(ArduinoIoTClient);
    const propertiesAPI = new ArduinoIoTApi.PropertiesV2Api(ArduinoIoTClient);

    return thingsApi.thingsV2List()
      .then(things => {
        things.forEach(thing => {
          propertiesAPI.propertiesV2List(thing.id)
            .then(properties => {
              properties.forEach(property => {
                ArduinoIoTCloud.onPropertyValue(thing.id, property.variable_name,
                  value => console.log(property.variable_name + ": " + value))
                  .then(() => console.log("Callback registered for " + property.variable_name))
                  .catch(error => console.log(error));
              });
            })
            .catch(error => console.log(error));
        });
      });
  })
  .catch(error => console.log(error));
```
